_posts/2024-06-25-create-your-first-python-package-scipy-2024.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: single
+title: "pyOpenSci Workshop: Create Your First Python Package"
+excerpt: "pyOpenSci will be running our `Create Your First Python Package: Make Your Python Code Easier to Share and Use
+` workshop at SciPy 2024. Read on to learn more!"
+author: "Jesse Mostipak"
+permalink: /blog/pyos-workshop-scipy-2024.html
+header:
+  overlay_image: images/blog/2024/june/scipy-2024-workshop.png
+  overlay_filter: rgba(20, 13, 36, 0.8)
+categories:
+  - blog-post
+  - community
+classes: wide
+toc: true
+comments: true
+---
+## Create Your First Python Package: Make Your Python Code Easier to Share and Use
+
+* **What:** A hands-on workshop, titled: [Create Your First Python Package: Make Your Python Code Easier to Share and Use](https://cfp.scipy.org/2024/talk/QT9GBY/)
+* **Where:** [SciPy 2024](https://www.scipy2024.scipy.org/), Room 316
+* **When:** Tuesday, July 9th, 2024, from 13:30--17:30 Pacific
+
+## Event overview
+
+Creating code that can be shared and reused is the gold-standard of open science. But the tools and skills to share your code can be tricky to learn. In this hands-on tutorial, you’ll learn how to turn your code into an installable Python module (a file containing Python code that defines functions, classes, and variables), that can be shared with others. We will provide pre-built code and module examples for each step of the tutorial to help make it more beginner-friendly, but you will need some basic Python skills to get the most out of this session.
+
+You will leave this tutorial understanding how to:
+* Create code that can be installed into different environments
+* Use [Hatch](https://hatch.pypa.io/latest/) as a workflow tool, making setup and installation of your code easier
+* Use Hatch to publish your package to the test version of [PyPI](https://pypi.org/)
+
+## What you need to know
+* Basic Python programming
+* How to write functions in Python
+* How to use a Python environment manager of your choosing
+
+## What you need installed
+* Python
+* An environment manager
+* Hatch
+
+### Install Python using `conda` and the `conda-forge` channel
```

Comment:
can you please remove the install python section? i think that was in that issue that i linked in the asana task maybe but it was in my very first comment. hatch will install python for them! yay! so we can remove it. 

i took your google doc and added to our github repo here: 

https://github.com/pyOpenSci/code-to-module-workshop/tree/main so you can essentially reuse the info in that markdown file here (it's almost the same but i removed the python piece.

Reply:
done. I copied and pasted your content from [https://github.com/pyOpenSci/code-to-module-workshop/tree/main](https://github.com/pyOpenSci/code-to-module-workshop/tree/main), starting at the Hatch & Python section, and going all the way through to, and including, the Useful Commands. let me know if you want that changed.